tests/test_docker_compose_validation.py

```
@@ -0,0 +1,232 @@
+        "DEVELOPMENT_MODE": "false",
+        "HOT_RELOAD": "true",
+        "RATE_LIMIT_PER_SECOND": "50.0",
+        "SLOW_REQUEST_THRESHOLD_MS": "5000.0",
+    }
+    for key, default in defaults.items():
+        assert f'${{{key}:-{default}}}' in text, f"Default for {key} should be {default}"
+
+def test_security_related_mounts_are_read_only():
+    text, _ = _read_compose_text()
+    assert "/home/dockermcp/.ssh:ro" in text, "SSH keys mount must be read-only"
+    assert "/var/run/docker.sock:/var/run/docker.sock:ro" in text, "Docker socket should be read-only to minimize risk"
```

Ruff (0.12.2)

No newline at end of file

Add trailing newline

(W292)